src/components/shapes/helpers.js

```diff
@@ -79,13 +85,21 @@ exports.getDataToPixel = function(gd, axis, isVertical) {
     return dataToPixel;
 };
 
-exports.getPixelToData = function(gd, axis, isVertical) {
+exports.getPixelToData = function(gd, axis, isVertical, opt) {
     var gs = gd._fullLayout._size;
     var pixelToData;
 
     if(axis) {
-        var r2d = exports.rangeToShapePosition(axis);
-        pixelToData = function(p) { return r2d(axis.p2r(p - axis._offset)); };
+        if(opt === 'domain') {
+            pixelToData = function(p) {
+                return ((isVertical ?
+                        (1 - (p - axis._offset) / axis._length) :
+                        (p - axis._offset) / axis._length));
+            };
```

Comment:
Let's use a temporary variable to avoid repetition.
```js
pixelToData = function(p) {
    var q = (p - axis._offset) / axis._length);
    return isVertical ? (1 - q) : q;
};
```

Reply:
d70f4e9c79bb1e8bd5a28002847c60b502da6884 fixes